### PR TITLE
Small code and typescript definition fixes.

### PR DESCRIPTION
**Pull request by EduFrazao**
- Fix IDBConnection:
   executeSql will fail to invoke "ok" callback sql execution fails
 - Typescript definition fixes to represent the new status
   and error message properties

### CODE REVIEW

File: src/index.ts [hunk 1]
```diff
@@ -1,22 +1,26 @@
 // IMPORTANT!!!!!!!!!!!
 // JSI BINDINGS DO NOT WORK WHEN CONNECTED TO THE DEBUGGER
 // This is a low level api
+type QueryResult = {
+  rows?: { // if status is undefined or 0 this object will contain the query results
```
Comment (ospfranco) [inline src/index.ts:5]: These would be great as JSDoc comments

File: src/index.ts [hunk 2]
```diff
@@ -58,9 +62,10 @@ export const openDatabase = (
         try {
           // console.warn(`[react-native-quick-sqlite], sql: `, sql, ` params: ` , params);
           let response = sqlite.executeSql(options.name, sql, params);
-          // enhance object to allow the sqlite-storage typeorm driver to work
-          response.rows.item = (idx: number) => response.rows._array[idx];
-
+          if(response.rows) {
+            // enhance object to allow the sqlite-storage typeorm driver to work
+            response.rows.item = (idx: number) => response.rows._array[idx];
```
Comment (ospfranco) [inline src/index.ts:67]: use optional chaining `response.rows?.item`

**Review by ospfranco — CHANGES_REQUESTED** [review]

File: src/index.ts (hunk 2 above)
Comment (EduFrazao) [reply src/index.ts:67]: Should I use this instead?
```
response.rows!.item = ()...
```
Using "response.rows?." are causing some lint errors.

Comment (EduFrazao) [reply src/index.ts:67]: When I try to use the optional chaining, I got this error:

```
src/index.ts: Invalid left-hand side in assignment expression. (88:10)
```

Using "response.rows!.item" fails too, with the system trying to assign the item function to a non existent response.rows object. On my tests, for now, only this check works as expected

```
if (response.rows) {
    ...
}
```

Comment (ospfranco) [reply src/index.ts:88]: I forgot the rows item is an object

**Comment by ospfranco** [comment]
I mostly copied your changes and applied them directly to main branch, also moved around the JSDoc comments, thanks for the PR!

**Comment by EduFrazao** [comment]
> I mostly copied your changes and applied them directly to main branch, also moved around the JSDoc comments, thanks for the PR!

Very nice. Thank you very mutch @ospfranco

**Pull request closed without merging**